packages/react-instantsearch-hooks-server/src/getServerState.tsx

```diff
@@ -0,0 +1,120 @@
+import { isIndexWidget } from 'instantsearch.js/es/widgets/index/index';
+import React from 'react';
+import ReactDOM from 'react-dom/server';
+
+import { InstantSearchServerContext } from 'react-instantsearch-hooks';
+
+import type {
+  InstantSearchServerApi,
+  InstantSearchSSRClient,
+  InstantSearchServerState,
+} from 'react-instantsearch-hooks';
+import type { InitialResults, InstantSearch } from 'instantsearch.js';
+import type { IndexWidget } from 'instantsearch.js/es/widgets/index/index';
+import type { ReactNode } from 'react';
+
+/**
+ * Returns the InstantSearch server state from a component.
+ */
+export async function getServerState(
+  children: ReactNode
+): Promise<InstantSearchServerState> {
+  const ssrClient: InstantSearchSSRClient = {
```

Comment:
isn't this more like a random object than a search client? Maybe something like this makes more sense:

```
const instantSearchRef = { current: undefined }

// L26
instantSearchRef.current = instantSearchInstance
```

Reply:
This "client" naming doesn't refer to a search client, but rather to the API to communicate between server contexts. "client" may not be a great name though. At first I went for `ssrContext` but that's also confusing because of React Context.

Perhaps `ssrApi` is enough – it remains internal anyway.

Do you have other suggestions?

Reply:
Oh, I better understand your suggestion now. The thing is that it won't only be the InstantSearch ref soon, but also the search client cache.

Reply:
ssrApi might work as well, but as said it should be easy enough to access any properties we need from the InstantSearch instance, therefore a InstantSearchRef / Box, or even simply a non-boxed variable which is mutated would work for me

Reply:
Addressed in ae76e8675eb98af797b9303649b158e130b80b57. I also removed the `InstantSearchSSRClient` type altogether.